### ISSUE TITLE
Driver not working on new kernel

### ISSUE THREAD

**Issue by enzocandaten** [nlkguy/archer-t2u-plus-linux#35]
My wifi adapter worked fine but after the kernel update it just shows the wifi connection as available but the adapter doesnt see any networks

![Captura de tela de 2024-06-20 14-27-39](https://github.com/nlkguy/archer-t2u-plus-linux/assets/135618311/d9992351-1568-4828-8eb8-15e9673d59c3)

I'm trying to use iwd to see if anything appears when i scan the networks with the device but i have the same result. Important to mention that iwd was enabled and running when i took these screenshots

![Captura de tela de 2024-06-20 14-41-50](https://github.com/nlkguy/archer-t2u-plus-linux/assets/135618311/208b3c6a-278f-497e-8006-7cc98dc25536)

![Captura de tela de 2024-06-20 14-42-17](https://github.com/nlkguy/archer-t2u-plus-linux/assets/135618311/5450a49f-6845-4057-b073-5a81d9165b96)

![Captura de tela de 2024-06-20 14-43-02](https://github.com/nlkguy/archer-t2u-plus-linux/assets/135618311/f9821e48-bf06-4eaa-9310-4b1dbdd29c35)

Keep it in mind that i have at least 5 networks that appear on other devices that dont run linux

**Comment by Mohammed-Yasin-Mulla** [comment]
Having the same issue here, after upgrading from Ubuntu 22 LST to Ubunut 24 LST

**Comment by Codethrix** [comment]
i am also facing the same issue with my both realtek chipset driver anyone knows how to fix this problem?

**Comment by MoistNutz** [comment]
I wouldn’t use that anymore it’s under scrutiny no one cares to read that
in bold letters it has malicious stuff on it no joke

**Comment by SuryaNath24** [comment]
I am facing the same issue.. tried to resolve it but it didn't work...

**Comment by S1D-D3F3ND3R** [comment]
Yeah, I'm encountering the exact issue after the update. Whenever I try to use sudo without wifi, the host doesn't show up. @nlkguy plz help

**Comment by MoistNutz** [comment]
Stop using it please for the love of god.

**Comment by S1D-D3F3ND3R** [comment]
@MoistNutz i purchased this adaptor 2 month ago I thought this is the best and budget friendly wifi adaptor now facing this issue after update

**Comment by MoistNutz** [comment]
It’s under scrutiny dude it gave me 184 Trojans from this .. get rid of it
I promise you

**Comment by enzocandaten** [comment]
@MoistNutz can you please send me the link to that? I work at a university and this is the default driver we use for all our computers that have a wi-fi module, if this is true i am very fucked

**Comment by S1D-D3F3ND3R** [comment]
> @MoistNutz can you please send me the link to that? I work at a university and this is the default driver we use for all our computers that have a wi-fi module, if this is true I am very fucked

The Wi-Fi adaptor works in Windows it doesn't need any additional driver for that the problem we are facing is with linux new kernel is not supporting this adaptor

**Comment by MoistNutz** [comment]
I’m absolutely not joking I had to wipe my computer, re install windows, I
had android.exploit.hack Trojans on my computer I was on Kali and installed
that adaptor on here and it worked but like I said I got on one day and had
182 Trojans it was insane and I don’t do crazy stuff just basic college
computer pen testing but you could be fine. But it also says “this is under
scrutiny” on the main thread lol

**Comment by enzocandaten** [comment]
@S1D-D3F3ND3R yes, but on windows machines we dont need to use this because we have the physical cd with the driver inside it and loads of drivers on the internet, but for linux machines tp-link stopped doing the drivers on kernel 4.16 i think, i was just trying to be sure that @MoistNutz is right because i use this fork from aircrack-ng repo on my workplace. Thanks to all of you for answering my thread, im just hoping that @nlkguy answers it too lmao

**Comment by MoistNutz** [comment]
You do when you run vms..

**Comment by S1D-D3F3ND3R** [comment]
I will make a live bootable drive with an older version of Kali Linux and use the adapter until the new drivers are available. If it works, I'll let you know. I will also use that bootable USB in VMware to avoid harming my main system, as @MoistNutz mentioned it installed a trojan.

**Comment by enzocandaten** [comment]
as i mentioned before i work at a university and both our antiviruses and firewall have not detected any trojans, maybe @MoistNutz installed something that harmed his pc after installing the driver? anyway, checking out to another commit works i guess, so you can to this while fixes arent made
_sudo apt update
sudo apt install dkms git
git clone -b v5.6.4.2 https://github.com/aircrack-ng/rtl8812au.git
cd rtl8812au
git checkout 63cf0b4
sudo make dkms_install_

**Comment by MoistNutz** [comment]
Sorry not responding this anymore a scam

**Comment by MoistNutz** [comment]
And I’m fucking sorry for the people who compiled this
